store/postgres/src/relational_queries.rs

```diff
@@ -803,6 +803,165 @@ impl<'a> QueryFragment<Pg> for PrefixComparison<'a> {
+            And(filters) | Or(filters) => {
+                for filter in filters {
+                    self.extend_with_filter(filter, parent_table, layout)?;
+                }
+            }
+            Child(field_name, entity, child_filter) => {
+                let child_table = layout.table_for_entity(entity)?;
+
+                self.join(parent_table, child_table, field_name)?;
+                self.extend_with_filter(child_filter, child_table, layout)?;
+            }
+            _ => {}
+        }
+        Ok(())
+    }
+
+    fn find(&self, parent_table: &'a Table, child_table: &'a Table) -> Option<&TableJoin> {
+        self.joins
+            .iter()
+            .filter_map(|join| {
+                if join.parent.name == parent_table.name && join.child.name == child_table.name {
+                    Some(join)
+                } else {
+                    None
+                }
+            })
+            .next()
+    }
+
+    fn get_or_add_prefix(&mut self, table: &'a Table) -> String {
+        return match self.prefixes.get(table.name.as_str()) {
+            Some(prefix) => prefix.to_string(),
+            None => {
+                let prefix = format!("c{}", self.prefixes.len());
+                self.prefixes.insert(table.name.to_string(), prefix.clone());
+                prefix
+            }
+        };
+    }
+
+    pub fn get_prefix(&self, table: &'a Table) -> String {
+        self.prefixes
+            .get(table.name.as_str())
+            .expect("table prefix not found")
+            .to_string()
+    }
+
+    fn join(
+        &mut self,
+        parent_table: &'a Table,
+        child_table: &'a Table,
+        field_name: &String,
+    ) -> Result<(), StoreError> {
+        Ok(match self.find(parent_table, child_table) {
+            None => {
+                let parent_prefix = self.get_or_add_prefix(parent_table);
+                let child_prefix = self.get_or_add_prefix(child_table);
+
+                self.joins.push(TableJoin {
+                    parent: parent_table,
+                    child: child_table,
+                    parent_prefix,
+                    child_prefix,
+                    parent_column: parent_table.column_for_field(field_name)?,
+                    child_column: child_table.primary_key(),
+                });
+            }
+            Some(_) => {}
+        })
+    }
+}
+
+impl<'a> QueryFragment<Pg> for TableJoinCollection<'a> {
+    fn walk_ast(&self, mut out: AstPass<Pg>) -> QueryResult<()> {
+        out.unsafe_to_cache_prepared();
+
+        fn push_column_with_prefix(
+            mut out: AstPass<Pg>,
+            column: &str,
+            prefix: &String,
+        ) -> QueryResult<()> {
+            out.push_sql(format!("{}.", prefix).as_str());
+            out.push_identifier(column)?;
+
+            Ok(())
+        }
+
+        for table_join in &self.joins {
+            out.push_sql("\n");
+
+            out.push_sql("left join ");
```

Comment:
you do want an `inner join` here; `left join` is shorthand for `left outer join`, and we want to suppress all parent rows that have no matching children.

Reply:
Thanks, used `inner join`.